File: java/src/test/java/testcases/chapterSix/ValidateSupplierProductTest.java
package testcases.chapterSix;

import org.assertj.core.api.Assertions;
import org.openqa.selenium.By;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.ui.Select;
import org.testng.annotations.Test;
import testcases.TestShopScenario;

import java.util.List;

public class ValidateSupplierProductTest extends TestShopScenario {

    @Test
    public void validateSupplierProduct(){

        driver.manage().window().maximize();

        Select dropdown = new Select(driver.findElement(By.cssSelector("select[name='supplier_list']")));
        dropdown.selectByVisibleText("AppleStore");

        List<WebElement> supplierProducts = driver.findElements(By.cssSelector("h5[itemprop='name']>a"));

        //option - bit messy but works
        boolean macBookFound = false;
        for(int i = 0; i < supplierProducts.size(); i++){
            if (supplierProducts.get(i).getText().equals("MacBook Air")){
                macBookFound = true;
                break;
            }
        }

        Assertions.assertThat(macBookFound)
                .as("Check if MacBook Air is in the list")
                .isTrue();

        driver.quit();
    }
}

File: cs/BootCampCode/testcases/chapterSix/ValidateSupplierProductTest.cs
using System;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using OpenQA.Selenium;
using OpenQA.Selenium.Chrome;
using OpenQA.Selenium.Support.UI;
using System.Collections.Generic;

namespace BootCampCode.testcases.chapterSix
{
    [TestClass]
    public class ValidateSupplierProductTest
    {
        [TestMethod]
        public void ValidateSupplierProduct()
        {
            IWebDriver driver = new ChromeDriver();

            WebDriverWait myWaitVar = new WebDriverWait(driver, TimeSpan.FromSeconds(20));

            // Open the website
            driver.Navigate().GoToUrl("https://techblog.polteq.com/testshop/index.php");

            driver.Manage().Window.Maximize();

            var dropdown = new SelectElement(driver.FindElement(By.CssSelector("select[name='supplier_list']")));
            dropdown.SelectByText("AppleStore");

            IList<IWebElement> supplierProducts = driver.FindElements(by: By.CssSelector("h5[itemprop='name']>a"));

            //option - bit messy but works
            Boolean macBookFound = false;
            for (int i = 0; i < supplierProducts.Count; i++)
            {
                if (supplierProducts[i].Text.Equals("MacBook Air"))
                {
                    macBookFound = true;
                    break;
                }
            }

            Assert.IsTrue(macBookFound, "Check if MacBook Air is in the list");

            driver.Quit();
        }
    }
}
